drivers/wifi/nrf700x/src/fw_load.c

```diff
@@ -22,36 +21,72 @@
 #include <zephyr/drivers/flash/nrf_qspi_nor.h>
 #endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH */
 
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(wifi_nrf, CONFIG_WIFI_NRF700X_LOG_LEVEL);
+
 #include <fmac_main.h>
-#include <rpu_fw_patches.h>
+
+/* INCBIN macro Taken from https://gist.github.com/mmozeiko/ed9655cf50341553d282 */
+#define STR2(x) #x
+#define STR(x) STR2(x)
+
+#ifdef __APPLE__
+#define USTR(x) "_" STR(x)
+#else
+#define USTR(x) STR(x)
+#endif
+
+#ifdef _WIN32
+#define INCBIN_SECTION ".rdata, \"dr\""
+#elif defined __APPLE__
+#define INCBIN_SECTION "__TEXT,__const"
+#else
+#define INCBIN_SECTION ".rodata"
+#endif
+
+/* this aligns start address to 16 and terminates byte array with explicit 0
+ * which is not really needed, feel free to change it to whatever you want/need
+ */
+#define INCBIN(prefix, name, file) \
+	__asm__(".section " INCBIN_SECTION "\n" \
+			".global " USTR(prefix) "_" STR(name) "_start\n" \
+			".balign 16\n" \
+			USTR(prefix) "_" STR(name) "_start:\n" \
+			".incbin \"" file "\"\n" \
+			\
+			".global " STR(prefix) "_" STR(name) "_end\n" \
+			".balign 1\n" \
+			USTR(prefix) "_" STR(name) "_end:\n" \
+			".byte 0\n" \
+	); \
+	extern __aligned(16)    const char prefix ## _ ## name ## _start[]; \
+	extern                  const char prefix ## _ ## name ## _end[];
+
+INCBIN(_bin, nrf70_fw, STR(CONFIG_NRF_WIFI_FW_BIN));
 
 enum nrf_wifi_status nrf_wifi_fw_load(void *rpu_ctx)
 {
 	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
-	struct nrf_wifi_fmac_fw_info fw_info;
+	struct nrf_wifi_fmac_fw_info fw_info = { 0 };
 #if defined(CONFIG_NRF_WIFI_PATCHES_EXT_FLASH) && defined(CONFIG_NORDIC_QSPI_NOR)
 	const struct device *flash_dev = DEVICE_DT_GET(DT_INST(0, nordic_qspi_nor));
 #endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH */
 
-	memset(&fw_info, 0, sizeof(fw_info));
-	fw_info.lmac_patch_pri.data = (void *) nrf_wifi_lmac_patch_pri_bimg;
-	fw_info.lmac_patch_pri.size = sizeof(nrf_wifi_lmac_patch_pri_bimg);
-	fw_info.lmac_patch_sec.data = (void *) nrf_wifi_lmac_patch_sec_bin;
-	fw_info.lmac_patch_sec.size = sizeof(nrf_wifi_lmac_patch_sec_bin);
-	fw_info.umac_patch_pri.data = (void *) nrf_wifi_umac_patch_pri_bimg;
-	fw_info.umac_patch_pri.size = sizeof(nrf_wifi_umac_patch_pri_bimg);
-	fw_info.umac_patch_sec.data = (void *) nrf_wifi_umac_patch_sec_bin;
-	fw_info.umac_patch_sec.size = sizeof(nrf_wifi_umac_patch_sec_bin);
+	status = nrf_wifi_fmac_fw_parse(rpu_ctx, (const uint8_t *)_bin_nrf70_fw_start,
```

Comment:
Where are _bin_nrf70_fw_start and _bin_nrf70_fw_end defined?

Reply:
They are defined by a simple `INCBIN` macro :) .